netsync/manager.go

```diff
@@ -406,9 +406,30 @@ func (sm *SyncManager) isSyncCandidate(peer *peerpkg.Peer) bool {
 			log.Errorf("Unable to query for segwit "+
 				"soft-fork state: %v", err)
 		}
+		if segwitActive && !peer.IsWitnessEnabled() {
```

Comment:
review nit: this is a nice refactor, would be even better if the change (breaking up the if conditions) was done in its own commit (easier to review) :)

Reply:
Would this still be needed with all the above changes?